src/main/kotlin/com/avail/descriptor/representation/AbstractDescriptor.kt

```diff
@@ -0,0 +1,4202 @@
+	 * [unsupported&#32;operation&#32;exception][AvailUnsupportedOperationException]
+	 * suitable to be thrown by the sender.
+	 *
+	 * The exception indicates that the receiver does not meaningfully implement
+	 * the method that immediately invoked this.  This is a strong indication
+	 * that the wrong kind of object is being used somewhere.
+	 *
+	 * This is a variant on [unsupportedOperation] to support Java
+	 * implementations of [AbstractDescriptor] until porting of the hierarchy is
+	 * finished. It should be thrown by Java callers, which cannot correctly
+	 * reason about flow around non-returning functions.
+	 *
+	 * @return
+	 *   Never returns anything; always throws.
+	 * @throws AvailUnsupportedOperationException
+	 *   Always.
+	 */
+	fun unsupportedOperationException (): AvailUnsupportedOperationException
+	{
+		val callerName =
+			try
+			{
+				throw Exception("just want the caller's frame")
+			}
+			catch (e: Exception)
+			{
+				e.stackTrace[1].methodName
+			}
+		throw AvailUnsupportedOperationException(javaClass, callerName)
+	}
+
+	/**
+	 * Answer whether the [argument types][AvailObject.argsTupleType] supported
+	 * by the specified [function type][FunctionTypeDescriptor] are acceptable
+	 * argument types for invoking a [function][FunctionDescriptor] whose type
+	 * is the self.
+	 *
+	 * @param self
+	 *   A function type.
+	 * @param functionType
+	 *   A function type.
+	 * @return
+	 *   `true` if the arguments of self are, pairwise, more general than those
+	 *   of `functionType`, `false` otherwise.
+	 * @see AvailObject.acceptsArgTypesFromFunctionType
+	 */
+	abstract fun o_AcceptsArgTypesFromFunctionType (
+		self: AvailObject,
+		functionType: A_Type): Boolean
+
+	/**
+	 * Answer whether these are acceptable [argument types][TypeDescriptor] for
+	 * invoking a [function][FunctionDescriptor] whose type is the self.
+	 *
+	 * @param self
+	 *   The receiver.
+	 * @param argTypes
+	 *   A list containing the argument types to be checked.
+	 * @return
+	 *   `true` if the arguments of the receiver are, pairwise, more general
+	 *   than those within the `argTypes` list, `false` otherwise.
+	 * @see AvailObject.acceptsListOfArgTypes
+	 */
+	abstract fun o_AcceptsListOfArgTypes (
+		self: AvailObject,
+		argTypes: List<A_Type>): Boolean
+
+	/**
+	 * Answer whether these are acceptable arguments for invoking a
+	 * [function][FunctionDescriptor] whose type is the self.
+	 *
+	 * @param self
+	 *   The receiver.
+	 * @param argValues
+	 *   A list containing the argument values to be checked.
+	 * @return
+	 *   `true` if the arguments of the receiver are, pairwise, more general
+	 *   than the types of the values within the `argValues` list, `false`
+	 *   otherwise.
+	 * @see AvailObject.acceptsListOfArgValues
+	 */
+	abstract fun o_AcceptsListOfArgValues (
+		self: AvailObject,
+		argValues: List<A_BasicObject>): Boolean
+
+	/**
+	 * Answer whether these are acceptable [argument types][TypeDescriptor] for
```

Comment:
`&#32;`

Reply:
(and more below)

Reply:
I fixed these systematically everywhere, which resulted in another 400+ changes. I use a few regular expressions and iterated to a fixed point, manually excluding false positives. I handled all links broken across lines. I reflowed all long comments. I will double check everything before I commit.